backend/src/main/java/ch/puzzle/pcts/service/validation/MemberValidationService.java

```diff
@@ -0,0 +1,77 @@
+package ch.puzzle.pcts.service.validation;
+
+import ch.puzzle.pcts.exception.PCTSException;
+import ch.puzzle.pcts.model.error.ErrorKey;
+import ch.puzzle.pcts.model.member.Member;
+import ch.puzzle.pcts.service.persistence.MemberPersistenceService;
+import jakarta.validation.ConstraintViolation;
+import jakarta.validation.Validation;
+import jakarta.validation.Validator;
+import jakarta.validation.ValidatorFactory;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Component;
+
+@Component
+
+public class MemberValidationService {
+
+    private final Validator validator;
+    private final MemberPersistenceService persistenceService;
+
+    public MemberValidationService(MemberPersistenceService persistenceService) {
+        this.persistenceService = persistenceService;
+        try (ValidatorFactory factory = Validation
+                .byDefaultProvider()
+                .configure()
+                .messageInterpolator(new FieldAwareMessageInterpolator(Validation
+                        .byDefaultProvider()
+                        .configure()
+                        .getDefaultMessageInterpolator()))
+                .buildValidatorFactory()) {
+            validator = factory.getValidator();
+        }
+    }
+
+    public void validateOnGetById(Long id) {
+        throwExceptionWhenIdIsNull(id);
+    }
+
+    public void validateOnCreate(Member member) {
+        validate(member);
```

Comment:
@MasterEvarior now in the Validation service to ensure that an Entity is indeed valid you just call validate and pass the member, then we invoke the beanValidater later on. This would potentially allows us to format the error in a special error dto and pass that to the frontend where you have all information you want and translate it properly there.